### PR TITLE
Attaching extra artifacts to bundle

### PR DESCRIPTION
**Pull request by ilkondr**
This patch implements ability to add extra artifacts to your bundle using recipe description (as it done for upgrade slots). These extra artifacts can be used as additional pre- and post-install logic providers or something else as/if you like

### CODE REVIEW

**Comment by ejoerns** [comment]
First of all thanks for you contribution! Did you test this on morty only or also on current poky releases?

Note that the two patches must be squashed for giving them a chance of being merged as they add and remove the same things. Is your intention to have them into morty branch, too?.

**Comment by ilkondr** [comment]
Unfortunately yes, this is tested on morty _only_ at the moment.
I don't expect these changes in morty branch as it is not maintained (or is it?)

**Comment by ilkondr** [comment]
tested on the latest _sumo_ and fixed `[depends]` flag handling

**Comment by yadutaf** [comment]
Nice addition ! I can definitely find use cases for that :+1: . Would it be possible to support a space delimited list of files in ``RAUC_BUNDLE_EXTRA_*`` ? That would allow supporting recipes producing more than one artifact.

File: classes/bundle.bbclass [hunk 1]
```diff
@@ -193,6 +206,26 @@ do_unpack_append() {
...
+
+    bundle_path = d.expand("${S}")
+    if not os.path.exists(bundle_path):
+        raise bb.build.FuncFailed('Failed creating symlink to %s' % imgname)
+    for extra in (d.getVar('RAUC_BUNDLE_EXTRAS') or "").split():
+        extraflags = d.getVarFlags('RAUC_EXTRA_%s' % extra)
+        extratype = extraflags.get('type') if extraflags and 'type' in extraflags else None
+        if not extratype:
+            bb.debug(1, "No type given for extra '%s', defaulting to 'file'" % extra)
+            extratype = 'file'
```
[the remark this answers is not in the document]
Comment (ilkondr) [reply classes/bundle.bbclass]: Agree, but I'd like to leave this debug message in place. So this kind of "internal magic" will be traced to log file

File: classes/bundle.bbclass [hunk 2]
```diff
@@ -193,6 +206,26 @@ do_unpack_append() {
...
+        if extratype == 'file':
+            extrafilename = extraflags.get('file') if extraflags else None
+            if not extrafilename:
+                bb.error("No file set for extra '%s'. Please specify via RAUC_EXTRA_%s[file]" % (extra,extra))
+                return
+            extrasrc = d.expand("${DEPLOY_DIR_IMAGE}/%s") % extrafilename
+            bundle_extrapath = '%s/%s' % (bundle_path, extrafilename)
+            shutil.copy(extrasrc, bundle_extrapath)
+            st = os.stat(extrasrc)
+            os.chmod(bundle_extrapath, st.st_mode | stat.S_IEXEC)
```
[the remark this answers is not in the document]
Comment (ilkondr) [reply classes/bundle.bbclass]: I leave this code for supporting multiple types as I was not sure single type is enough for me.
Now I understand that this single type covers all my needs. 
Maybe we should add something like "source" type that means extra atrifact is fetched via SRC_URI link.
Please feel free to suggest another types which you consider useful.

**Comment by ilkondr** [comment]
Added types 'artifact' which is default and 'source'.
Multiple files can now be listed in [files] flag.
Rebased on current master and force-pushed my changes.

**Comment by ejoerns** [comment]
FTR: I've proposed an alternative (more lightweight) implementation of extra artifact fetch in #102.

**Pull request closed without merging**